Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -189,7 +203,7 @@ private IPooledConnection Acquire(CancellationToken cancellationToken)
                 {
                     while (true)
                     {
-                        if (_disposeCalled)
+                        if (_closedMarker > 0)
```

Comment:
Maybe extract `_closedMarker > 0` in a method like `IsClosed()`?
It is used in many places and feels a little bit cryptic.